solidity/contracts/igps/InterchainGasPaymaster.sol

```diff
@@ -68,9 +74,34 @@ contract InterchainGasPaymaster is IInterchainGasPaymaster, OwnableUpgradeable {
             require(_success, "Interchain gas payment refund failed");
         }
 
-        emit GasPayment(_messageId, _gasAmount, msg.value);
+        emit GasPayment(_messageId, _gasAmount, _requiredPayment);
+    }
+
+    /**
+     * @notice Sets the gas oracle for a remote domain.
+     * @param _remoteDomain The remote domain.
+     * @param _gasOracle The gas oracle.
+     */
+    function setGasOracle(uint32 _remoteDomain, address _gasOracle)
```

Comment:
Do we want batch versions/turn this into a batch version?

Reply:
yeah we need this to use `upgradeAndCall` with the contract upgrade, adding